sdk/python/src/dagger/mod/_resolver.py

```diff
@@ -129,9 +131,15 @@ def register(self, typedef: dagger.TypeDef) -> dagger.TypeDef:
     def return_type(self):
         """Return the resolved return type of the wrapped function."""
         try:
-            return self._type_hints["return"]
+            r = self._type_hints["return"]
         except KeyError:
             return MissingType
+        if r in (typing.Self, typing_extensions.Self):
```

Comment:
I think it's also possible to return `list[Self]`, which should be converted to `list[Foo]`?

Reply:
While true, I'm not sure it's worth it for the added complexity since it opens a can of worms that's not trivial to support. What would be the use case?

There's more context in [PEP-0673](https://peps.python.org/pep-0673/) for why this feature was added. The main motivation was for a more intuitive way to type an inherited method (in a subclass) that `return self`. Subclassing isn't even tested in Dagger at the moment.

My motivation here was as a convenience because I don't like to use forward references like `-> "Foo"`, so doing `-> Self` is a way out of that, but you can still do `-> "list[Foo]"`.